src/sage/homology/graded_resolution.py

```diff
@@ -9,51 +9,52 @@
 
     sage: S.<x,y,z,w> = PolynomialRing(QQ)
     sage: I = S.ideal([y*w - z^2, -x*w + y*z, x*z - y^2])
-    sage: r = I.graded_free_resolution(algorithm='minimal')
-    sage: r
+    sage: r = I.graded_free_resolution(algorithm='minimal')                             # optional - sage.libs.singular
+    sage: r                                                                             # optional - sage.libs.singular
```

Comment:
Like the `free_resolution` module,  the `graded_resolution` module depends on `sage.libs.singular`. Hence file annotation is preferable.

Reply:
Thanks, done.